frontend/src/components/common/Table/Table.tsx

```diff
@@ -169,18 +172,27 @@ export default function Table<RowItem extends Record<string, any>>({
     return (tableProps.data ?? []).filter(it => filterFunction(it));
   }, [tableProps.data, filterFunction]);
 
-  const gridTemplateColumns = tableProps.columns
-    .filter(it => {
-      const isHidden = tableProps.state?.columnVisibility?.[it.id!] === false;
-      return !isHidden;
-    })
-    .map(it => {
-      if (typeof it.gridTemplate === 'number') {
-        return `${it.gridTemplate}fr`;
-      }
-      return it.gridTemplate ?? '1fr';
-    })
-    .join(' ');
+  const gridTemplateColumns = useMemo(() => {
+    let preGridTemplateColumns = tableProps.columns
+      .filter(it => {
+        const isHidden = tableProps.state?.columnVisibility?.[it.id!] === false;
+        return !isHidden;
+      })
+      .map(it => {
+        if (typeof it.gridTemplate === 'number') {
+          return `${it.gridTemplate}fr`;
+        }
+        return it.gridTemplate ?? '1fr';
+      })
+      .join(' ');
+    if (tableProps.enableRowActions) {
+      preGridTemplateColumns = `${preGridTemplateColumns} 0.05fr`;
+    }
+    if (tableProps.enableRowSelection) {
+      preGridTemplateColumns = `0.05fr ${preGridTemplateColumns}`;
+    }
+    return preGridTemplateColumns;
+  }, [tableProps]);
```

Comment:
tableProps will be different each render so this memo won't really work

Reply:
better?

Reply:
yes but I think `columnVisibility` is missing, could you try hiding/showing some columns to confirm it still works?

Reply:
This still worked.